src/clj/puppetlabs/services/jruby/jruby_puppet_internal.clj

```diff
@@ -96,6 +134,35 @@
     (.runScriptlet "require 'jar-dependencies'")
     (.runScriptlet "require 'puppet/server/master'")))
 
+(schema/defn ^:always-validate config->puppet-config :- HashMap
+  "Given the raw jruby-puppet configuration section, return a
+  HashMap with the configuration necessary for ruby Puppet."
+  [config :- jruby-schemas/JRubyPuppetConfig]
+  (let [puppet-config (new HashMap)]
+    (doseq [[setting dir] [[:master-conf-dir "confdir"]
+                           [:master-code-dir "codedir"]
```

Comment:
Probably don't need `master-code-dir`, `master-run-dir`, or `master-log-dir` for the stable branch, right?  Not sure if it hurts but the underlying Puppet Ruby code wouldn't support them.